Python/Product/Analysis/Interpreter/SharedDatabaseState.cs

```diff
@@ -34,7 +34,7 @@ class SharedDatabaseState : ITypeDatabaseReader {
 #endif
         private readonly List<IPythonModule> _modulesLoading = new List<IPythonModule>();
         private List<Action> _fixups;
-        private List<Action<IPythonType>> _objectTypeFixups;
+        private List<Action<IPythonType>> _objectTypeFixups = new List<Action<IPythonType>>();

```

Comment:
It's not entirely clear to me whether the field (as opposed to collection) can be changed from concurrent threads. Code that's writing to this field does so under a lock, but code that's reading it (null checks etc) normally doesn't lock, and so could result in stale reads without a memory barrier. It feels like the latter should be using Volatile.Read, or else the writes don't have to be under lock.